arch/xtensa/src/esp32s2/esp32s2_tim_lowerhalf.c

```diff
@@ -0,0 +1,595 @@
+/****************************************************************************
+ * arch/xtensa/src/esp32s2/esp32s2_tim_lowerhalf.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <sys/types.h>
+#include <stdbool.h>
+#include <string.h>
+#include <assert.h>
+#include <errno.h>
+#include <debug.h>
+#include <stdio.h>
+#include <stdint.h>
+
+#include <nuttx/arch.h>
+#include <nuttx/timers/timer.h>
+
+#include "hardware/esp32s2_soc.h"
+
+#include "esp32s2_tim.h"
+#include "esp32s2_clockconfig.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+struct esp32s2_timer_lowerhalf_s
+{
+  FAR const struct timer_ops_s *ops;        /* Lower half operations */
+  FAR struct esp32s2_tim_dev_s *tim;        /* esp32s2 timer driver */
+  tccb_t                        callback;   /* Interrupt callback */
+  FAR void                     *arg;        /* Argument passed to upper half callback */
+  bool                          started;    /* True: Timer has been started */
+  void *upper;                              /* Pointer to watchdog_upperhalf_s */
```

Comment:
```suggestion
  FAR void                     *upper;                              /* Pointer to watchdog_upperhalf_s */
```
nit: Just to follow the adopted alignment method.